server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java

```diff
@@ -1672,7 +1673,9 @@ private static void updateOrganizationBroker(RealmModel realm, IdentityProviderR
 
             String domain = representation.getConfig().get(OrganizationModel.ORGANIZATION_DOMAIN_ATTRIBUTE);
 
-            if (domain != null && org.getDomains().map(OrganizationDomainModel::getName).noneMatch(domain::equals)) {
+            if (StringUtil.isBlank(domain)) {
```

Comment:
Changed this to make sure empty strings remove the domain from the broker config.